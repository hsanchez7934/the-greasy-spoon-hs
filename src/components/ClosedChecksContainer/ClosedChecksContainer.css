#closedchecks-container {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  overflow: scroll; }

.closed-checks {
  border: 1px solid black;
  width: 50%;
  height: 20%;
  margin: 30px 0px;
  -webkit-box-shadow: 0px 10px 37px 5px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 0px 10px 37px 5px rgba(0, 0, 0, 0.75);
  box-shadow: 0px 10px 37px 5px rgba(0, 0, 0, 0.75);
  background: linear-gradient(0deg, #011f4b, #005b96 50%, #011f4b); }

.closed-check-top {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 50%; }

.closed-check-bottom {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 50%; }

.closed-check-date {
  height: 20%;
  width: 80%;
  text-align: center;
  font-size: 1em;
  margin: 0;
  color: #b3cde0; }

.closed-check-table-number {
  padding-top: 10px;
  height: 50%;
  width: 80%;
  text-align: center;
  font-size: 2em;
  margin: 0;
  color: #F7F7F5; }

a.see-details-button-link-tag {
  height: 40%;
  width: 25%; }

.see-details-button {
  background-color: #6497b1;
  border: 0px;
  height: 100%;
  width: 100%;
  font-size: 1.2em;
  color: #F7F7F5; }

.see-details-button:hover {
  background-color: #b3cde0;
  cursor: pointer;
  color: #011f4b; }

.no-closed-checks-warning {
  font-size: 3em; }
