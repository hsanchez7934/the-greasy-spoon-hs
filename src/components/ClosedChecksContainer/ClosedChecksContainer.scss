@import "../../styles/colors.scss";

#closedchecks-container {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  overflow: scroll;
}

.closed-checks {
  border: 1px solid black;
  width: 50%;
  height: 20%;
  margin: 30px 0px;
  -webkit-box-shadow: 0px 10px 37px 5px rgba(0,0,0,0.75);
  -moz-box-shadow: 0px 10px 37px 5px rgba(0,0,0,0.75);
  box-shadow: 0px 10px 37px 5px rgba(0,0,0,0.75);
  background: linear-gradient(0deg, $color1, $color3 50%, $color1);
}

.closed-check-top {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 50%;
}

.closed-check-bottom {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 50%;
}

.closed-check-date {
  height: 20%;
  width: 80%;
  text-align: center;
  font-size: 1em;
  margin: 0;
  color: $color5;
}

.closed-check-table-number {
  padding-top: 10px;
  height: 50%;
  width: 80%;
  text-align: center;
  font-size: 2em;
  margin: 0;
  color: $color6;
}

a.see-details-button-link-tag {
  height: 40%;
  width: 25%;
}

.see-details-button {
  background-color: $color4;
  border: 0px;
  height: 100%;
  width: 100%;
  font-size: 1.2em;
  color: $color6;
}

.see-details-button:hover {
  background-color: $color5;
  cursor: pointer;
  color: $color1;
}

.no-closed-checks-warning {
  font-size: 3em;
}
