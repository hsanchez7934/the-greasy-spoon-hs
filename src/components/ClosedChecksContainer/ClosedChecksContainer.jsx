import React, { Component } from 'react';
import './ClosedChecksContainer.css';
import { fetchTables, fetchCheckById, fetchChecks } from '../../actions';
import {
  findTable,
  formatDate,
  queryForChecks } from '../../helperFunctions.js';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

class ClosedChecksContainer extends Component {

  componentDidMount() {
    this.props.fetchTables();
    this.props.fetchChecks();
  }

  createClosedTablesCard = () => {
    return this.props.checks.map( (check, index) => {
      if (check.closed === true ) {
        return <article key={index} id={check.id} className='closed-checks'>
          <div className='closed-check-top'>
            <p className='closed-check-date'>
              {formatDate(check.dateCreated)}
            </p>
            <h1 className='closed-check-table-number'>
              Table {findTable(check.tableId, this.props.tables)}
            </h1>
          </div>
          <div className='closed-check-bottom'>
            <Link to='/currentcheck' className='see-details-button-link-tag'>
              <button
                onClick={() => this.props.fetchCheckById(check.id)}
                className='see-details-button'>
                See Details
              </button>
            </Link>
          </div>
        </article>;
      }
    });
  }

  render() {
    if (queryForChecks(this.props.checks, true).length === 0) {
      return (
        <section id='closedchecks-container'>
          <p className='no-closed-checks-warning'>NO CLOSED CHECKS</p>
        </section>
      );
    } else if (this.props.tables.length !== 0) {
      return (
        <section id='closedchecks-container'>
          {
            this.createClosedTablesCard()
          }
        </section>
      );
    } else {
      return (
        <div></div>
      );
    }
  }
}

ClosedChecksContainer.propTypes = {
  fetchChecks: PropTypes.func,
  fetchTables: PropTypes.func,
  tables: PropTypes.array,
  checks: PropTypes.array,
  fetchCheckById: PropTypes.func
};

const mapStateToProps = store => ({
  checks: store.checks,
  tables: store.tables
});

const mapDispatchToProps = dispatch => ({
  fetchChecks: () => dispatch(fetchChecks()),
  fetchTables: () => dispatch(fetchTables()),
  fetchCheckById: (id) => dispatch(fetchCheckById(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(ClosedChecksContainer);
