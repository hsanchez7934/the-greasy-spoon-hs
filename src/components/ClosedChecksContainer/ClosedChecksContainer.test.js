import React, { Component } from 'react';
import { ClosedChecksContainer } from './ClosedChecksContainer';
import { shallow } from 'enzyme';
import config from '../../tests/testSetup.js';
import toJson from 'enzyme-to-json';
import tables from '../../testMockData/tables.js';
import checksArray from '../../testMockData/checks.js';

describe(`ClosedChecksContainer unit tests`, () => {
  let wrapper;
  let func1;
  let func2;
  let func3;

  beforeEach(() => {
    func1 = () => {};
    func2 = () => {};
    func3 = () => {};
    wrapper = shallow(
      <ClosedChecksContainer
        tables={tables}
        checks={checksArray}
        fetchChecks={func1}
        fetchTables={func2}
        fetchCheckById={func3} />
    );
  });

  test(`should create an instance of ClosedChecksContainer`, () => {
    expect(wrapper.exists()).toEqual(true);
  });

  test(`should match snapshot`, () => {
    expect(wrapper).toMatchSnapshot();
  });

  test(`should render section node`, () => {
    const section = wrapper.find('section');

    expect(section.exists()).toEqual(true);
    expect(section.type()).toEqual('section');
  });

  test(`should render article that has closed check information`, () => {
    const article = wrapper.find('article');
    
    expect(article.exists()).toEqual(true);
    expect(article.type()).toEqual('article');
  });
});
