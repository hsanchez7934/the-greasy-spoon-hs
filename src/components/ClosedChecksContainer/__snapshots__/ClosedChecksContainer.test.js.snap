// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClosedChecksContainer unit tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ClosedChecksContainer
    checks={
        Array [
            Object {
              "closed": false,
              "createdBy": "23037775-39f4-48fa-806a-908727b77957",
              "dateCreated": "2018-02-13T02:26:08.268361Z",
              "dateUpdated": "2018-02-13T03:01:10.865959Z",
              "id": "6400dc63-857b-4c12-a1bd-469e4cccae12",
              "tableId": "2644ece3-83dd-4deb-ae02-54f4df083e16",
              "tax": null,
              "tip": null,
            },
            Object {
              "closed": false,
              "createdBy": "23037775-39f4-48fa-806a-908727b77957",
              "dateCreated": "2018-02-13T02:26:19.335022Z",
              "dateUpdated": "2018-02-13T03:01:01.897597Z",
              "id": "ab7a931f-2d77-4d87-b09a-737d196d54f6",
              "tableId": "51f719e1-830e-40bb-9c1a-493ccc13cbc0",
              "tax": null,
              "tip": null,
            },
            Object {
              "closed": true,
              "createdBy": "23037775-39f4-48fa-806a-908727b77957",
              "dateCreated": "2018-02-13T02:55:50.871818Z",
              "dateUpdated": "2018-02-13T02:56:03.891757Z",
              "id": "f2bf62e8-5dee-4609-b695-0c273318fb79",
              "tableId": "f5a3d871-1548-4be5-9f5a-e9a6e2011187",
              "tax": 4.6,
              "tip": 11.178,
            },
          ]
    }
    fetchCheckById={[Function]}
    fetchChecks={[Function]}
    fetchTables={[Function]}
    tables={
        Array [
            Object {
              "id": "2644ece3-83dd-4deb-ae02-54f4df083e16",
              "number": 1,
            },
            Object {
              "id": "c482731d-19a4-4d1f-90ab-e4dc4ac7d28d",
              "number": 2,
            },
            Object {
              "id": "51f719e1-830e-40bb-9c1a-493ccc13cbc0",
              "number": 3,
            },
            Object {
              "id": "56fe84ff-0655-4972-b2b0-b097e0a26ca1",
              "number": 4,
            },
            Object {
              "id": "f5a3d871-1548-4be5-9f5a-e9a6e2011187",
              "number": 5,
            },
            Object {
              "id": "cb336b2a-a16f-4734-8e62-f165d5a2ac03",
              "number": 6,
            },
            Object {
              "id": "5d6e6290-a0bb-4aa9-ba3d-9071e5a65a93",
              "number": 7,
            },
            Object {
              "id": "b0672e44-b959-4f09-ad7e-6f53a386d815",
              "number": 8,
            },
            Object {
              "id": "4cf15df2-c1bf-435f-8e8b-ad6aba48937c",
              "number": 9,
            },
            Object {
              "id": "31ccc746-4ade-43ea-add1-3dba513feb85",
              "number": 10,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
        <article
          className="closed-checks"
          id="f2bf62e8-5dee-4609-b695-0c273318fb79"
>
          <div
                    className="closed-check-top"
          >
                    <p
                              className="closed-check-date"
                    >
                              Monday, February 12th 2018, 7:55:50 pm
                    </p>
                    <h1
                              className="closed-check-table-number"
                    >
                              Table 
                              5
                    </h1>
          </div>
          <div
                    className="closed-check-bottom"
          >
                    <Link
                              className="see-details-button-link-tag"
                              replace={false}
                              to="/currentcheck"
                    >
                              <button
                                        className="see-details-button"
                                        onClick={[Function]}
                              >
                                        See Details
                              </button>
                    </Link>
          </div>
</article>,
      ],
      "id": "closedchecks-container",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="closed-check-top"
>
              <p
                            className="closed-check-date"
              >
                            Monday, February 12th 2018, 7:55:50 pm
              </p>
              <h1
                            className="closed-check-table-number"
              >
                            Table 
                            5
              </h1>
</div>,
            <div
              className="closed-check-bottom"
>
              <Link
                            className="see-details-button-link-tag"
                            replace={false}
                            to="/currentcheck"
              >
                            <button
                                          className="see-details-button"
                                          onClick={[Function]}
                            >
                                          See Details
                            </button>
              </Link>
</div>,
          ],
          "className": "closed-checks",
          "id": "f2bf62e8-5dee-4609-b695-0c273318fb79",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="closed-check-date"
>
                  Monday, February 12th 2018, 7:55:50 pm
</p>,
                <h1
                  className="closed-check-table-number"
>
                  Table 
                  5
</h1>,
              ],
              "className": "closed-check-top",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Monday, February 12th 2018, 7:55:50 pm",
                  "className": "closed-check-date",
                },
                "ref": null,
                "rendered": "Monday, February 12th 2018, 7:55:50 pm",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Table ",
                    5,
                  ],
                  "className": "closed-check-table-number",
                },
                "ref": null,
                "rendered": Array [
                  "Table ",
                  5,
                ],
                "type": "h1",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="see-details-button-link-tag"
                replace={false}
                to="/currentcheck"
>
                <button
                                className="see-details-button"
                                onClick={[Function]}
                >
                                See Details
                </button>
</Link>,
              "className": "closed-check-bottom",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="see-details-button"
                  onClick={[Function]}
>
                  See Details
</button>,
                "className": "see-details-button-link-tag",
                "replace": false,
                "to": "/currentcheck",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "See Details",
                  "className": "see-details-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "See Details",
                "type": "button",
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
          <article
            className="closed-checks"
            id="f2bf62e8-5dee-4609-b695-0c273318fb79"
>
            <div
                        className="closed-check-top"
            >
                        <p
                                    className="closed-check-date"
                        >
                                    Monday, February 12th 2018, 7:55:50 pm
                        </p>
                        <h1
                                    className="closed-check-table-number"
                        >
                                    Table 
                                    5
                        </h1>
            </div>
            <div
                        className="closed-check-bottom"
            >
                        <Link
                                    className="see-details-button-link-tag"
                                    replace={false}
                                    to="/currentcheck"
                        >
                                    <button
                                                className="see-details-button"
                                                onClick={[Function]}
                                    >
                                                See Details
                                    </button>
                        </Link>
            </div>
</article>,
        ],
        "id": "closedchecks-container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="closed-check-top"
>
                <p
                                className="closed-check-date"
                >
                                Monday, February 12th 2018, 7:55:50 pm
                </p>
                <h1
                                className="closed-check-table-number"
                >
                                Table 
                                5
                </h1>
</div>,
              <div
                className="closed-check-bottom"
>
                <Link
                                className="see-details-button-link-tag"
                                replace={false}
                                to="/currentcheck"
                >
                                <button
                                                className="see-details-button"
                                                onClick={[Function]}
                                >
                                                See Details
                                </button>
                </Link>
</div>,
            ],
            "className": "closed-checks",
            "id": "f2bf62e8-5dee-4609-b695-0c273318fb79",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="closed-check-date"
>
                    Monday, February 12th 2018, 7:55:50 pm
</p>,
                  <h1
                    className="closed-check-table-number"
>
                    Table 
                    5
</h1>,
                ],
                "className": "closed-check-top",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Monday, February 12th 2018, 7:55:50 pm",
                    "className": "closed-check-date",
                  },
                  "ref": null,
                  "rendered": "Monday, February 12th 2018, 7:55:50 pm",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Table ",
                      5,
                    ],
                    "className": "closed-check-table-number",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Table ",
                    5,
                  ],
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="see-details-button-link-tag"
                  replace={false}
                  to="/currentcheck"
>
                  <button
                                    className="see-details-button"
                                    onClick={[Function]}
                  >
                                    See Details
                  </button>
</Link>,
                "className": "closed-check-bottom",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="see-details-button"
                    onClick={[Function]}
>
                    See Details
</button>,
                  "className": "see-details-button-link-tag",
                  "replace": false,
                  "to": "/currentcheck",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "See Details",
                    "className": "see-details-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "See Details",
                  "type": "button",
                },
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
